processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -37,6 +37,8 @@
   public static final String DEFAULT_TIMEOUT_KEY = "defaultTimeout";
   @Deprecated
   public static final String CHUNK_PERIOD_KEY = "chunkPeriod";
+  public static final String NUM_BROKER_PARALLEL_COMBINE_THREADS = "numBrokerParallelCombineThreads";
```

Comment:
Is it possible to not add this configurations and figure out the most appropriate setting automatically?

Reply:
Hmm, what is the most appropriate setting you are thinking? I guess it's possible to automatically pick something reasonable based on some statistics like intermediate aggregate size, # of rows to be combined, etc. This is possible in the future when we have a system to collect such statistics.

It currently picks all available processing threads if this value is set to -1 (https://github.com/apache/incubator-druid/pull/6629/files#diff-d8ff07a1952a0070326e5647b4e738f9R45). Do you have a better idea?